### ISSUE TITLE
"Internal" folder takes a while to load

### ISSUE THREAD

**Issue by Elaborendum** [SimpleMobileTools/Simple-File-Manager#503]
When first opening the app, an empty "Root" folder appears for about 20 seconds or so before the Internal folder loads. My device isn't rooted, either.

You can go to another folder (like external storage) while waiting, and it will instantly load, but trying to go back to the internal folder will seemingly not work, as nothing really happens.

Once it finally loads, you can go to any subfolder, but if you try to go back it will get stuck loading again and won't show any sign that it's actually going back to the main folder.

The app still works while it's loading, I can go to the settings without issue and if I take long enough, it might be loaded when I go back.

Android 11 - SFM v 6.9.4

**Comment by tibbi** [comment]
Dont you have a lot of folders shown directly at the internal storage?

**Comment by Elaborendum** [comment]
No, in every occasion it takes a while for them to appear. Like I said, even at startup it only shows an empty "Root" folder before it's replaced by the proper Internal folder. Or what do you mean exactly?

**Comment by ThisIsPaulDaily** [comment]
I have long load times for switching to my SD card. I think it nay be related to Google's latest attempt at killing android SD card support with the scoped app storage requirements and the latest API. 

It also broke OSMAnd~ for those of us who keep map data on SD cards. 

Just right now, It takes about a minute to load my 256gb brand new SD card and if I click into a folder on it and go back up a level to root of the SD card it will take another minute to load. 

A solution discussed in OSM was to target the old api, but google won't let apps on the play store like it. 

Supposedly file managers are to be exempt from this issue, but it doesn't seem like it.

**Comment by ThisIsPaulDaily** [comment]
https://github.com/osmandapp/OsmAnd/issues/13292
See also^

**Comment by meganleewebb** [comment]
Had the same problem with excessively slow load times. Internal folder, and opening folders was taking about 30 seconds. 

I'm running Nova with sesame search. I found sesame cache size was 500 meg. Normally it's <7 meg. Cleared the cache, and folders opened instantly. (Ended up having to delete and reinstall sesame as the cache kept growing.) I have other file managers installed, but they didn't have the slow down. 

Maybe this helps someone else.

**Comment by lamyergeier** [comment]
@tibbi Facing same issue on Android 11 Samung One UI.
